src/react/providers/chat-room-provider.tsx

```diff
@@ -77,42 +128,104 @@ export const ChatRoomProvider: React.FC<ChatRoomProviderProps> = ({
 }) => {
   const client = useChatClient();
   const logger = useLogger();
-  logger.trace(`ChatRoomProvider();`, { roomId, options, release, attach });
+  logger.debug(`ChatRoomProvider();`, { roomId, options, release, attach });
 
-  const [value, setValue] = useState({ room: client.rooms.get(roomId, options) });
+  // Set the initial room promise, we do this in a function to avoid rooms.get being called
+  // every time the component re-renders
+  // In StrictMode this will be called twice one after the other, but that's ok
+  const [value, setValue] = useState<ChatRoomContextType>(() => {
+    logger.debug(`ChatRoomProvider(); initializing value`, { roomId, options });
+    return { room: client.rooms.get(roomId, options).catch(), roomId: roomId, options: options };
+  });
 
+  // Create an effect that changes the room when the id or options change
+  const prevId = useRef(roomId);
+  const prevOptions = useRef(options);
+  const roomReleaseQueue = useRef(new RoomReleaseQueue(logger));
+
+  // update the release queue if the logger changes - as it means we have a new client
+  // and only if it actually changes, not because strict mode ran it twice
   useEffect(() => {
-    const room = client.rooms.get(roomId, options);
+    // Don't create a new queue if the logger hasn't actually changed
+    if (roomReleaseQueue.current.logger === logger) {
+      return;
+    }
 
-    // Update state if room instance has changed.
-    setValue((prev) => {
-      if (prev.room === room) {
-        return prev;
-      }
-      return { room };
-    });
+    logger.debug(`ChatRoomProvider(); updating release queue`);
+    roomReleaseQueue.current = new RoomReleaseQueue(logger);
+  }, [logger]);
 
-    if (attach) {
-      // attachment error and/or room status is available via useRoom
-      // or room.status, no need to do anything with the promise here
-      logger.debug(`ChatRoomProvider(); attaching room`, { roomId });
-      void room.attach().catch(() => {
-        // Ignore, the error will be available via various room status properties
-      });
+  useEffect(() => {
+    if (prevId.current === roomId && prevOptions.current === options) {
```

Comment:
The rest of this should also run if the client changes even if the roomId and options stay the same?